catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -362,6 +371,38 @@ public boolean onOptionsItemSelected(MenuItem item) {
 		return true;
 	}
 
+	private void importDownloadedProject(@NonNull final Uri uri) {
+		new RequiresPermissionTask(REQUEST_PERMISSIONS_MAIN_STORAGE,
+				Arrays.asList(WRITE_EXTERNAL_STORAGE, READ_EXTERNAL_STORAGE),
+				R.string.runtime_permission_general) {
+			public void task() {
+				InputStream projectStream;
+				try {
+					projectStream = getContentResolver().openInputStream(uri);
+					setShowProgressBar(true);
+					ProjectImportFromArchiveTask.ProjectImportListener importListener = new ProjectImportFromArchiveTask.ProjectImportListener() {
+						@Override
+						public void onImportFinished(boolean success) {
+							setShowProgressBar(false);
+							if (success) {
+								ToastUtil.showSuccess(MainMenuActivity.this, R.string.project_successful_imported_toast);
+								Log.d("IMPORT_PROJECT", "Project imported successfully");
+							} else {
+								ToastUtil.showError(MainMenuActivity.this, R.string.error_during_importing_projects_toast);
+								Log.e("IMPORT_PROJECT", "Project import failed");
+							}
+						}
+					};
+					new ProjectImportFromArchiveTask(MainMenuActivity.this, importListener).execute(projectStream).get();
```

Comment:
get() is blocking and will lead this to wait for the task to complete. so if i understand this correctly, that makes that async task sync to the main thread, so while its executed in another thread, the main thread is stuck here until its finished, therefore defying the async in asynctask.
https://developer.android.com/reference/android/os/AsyncTask.html#get()

Reply:
I think it's probably unnecessary, that's true. But I highly doubt it's blocking the ui thread because it's executed from another async task.